### ISSUE TITLE
FuTIL imports and "thin" libraries

### ISSUE THREAD

**Issue by rachitnigam** [cucapra/futil#323]
FuTIL currently only allows `import` statements for FuTIL libraries. #268 tries to solve this by removing the distinction between libraries files (that can only contain `primitive` definitions) and FuTIL files (that can only contain `component` definitions). It allows arbitrary mixing of both kinds of definitions.

While we're at it, I think we should reconsider the idea of "fat" libraries where the primitive definitions embed the Verilog implementation inside them. Instead, we should use Dahlia style `import` statements that define the interface for primitives without implementing them.

A few reasons to do this:
1. Being able to put primitives inside `.sv` files and testing them separately from the compiler. Furthermore, in the future, we can try to automatically generate `primitive` interfaces from an `.sv` file.
2. We cannot manipulate the strings embedded in the library files. Why bother carrying them throughout the compilation process?

**Comment by sampsyo** [comment]
Yeah, this is probably a good idea. FWIW, I think a good general term for what this would look like is `extern` declarations… i.e., declarations without definitions. Then the programs would need to be "linked" after the fact.

Thinking back to why we went with the "fat library" approach, I believe we did it so we could do some limited metaprogramming on the SystemVerilog implementations, right? But I believe we now use SV `parameter`s for all those compile-time parameters… i.e., we rely on SV itself for metaprogramming rather than doing it ourselves. So the need has diminished. Does this sound about right?
